multiplication_models/functions_for_all_nutrients.py

```diff
@@ -0,0 +1,158 @@
+import pandas as pd, numpy as np
+from db_queries import get_ids, get_outputs, get_location_metadata, get_population, get_covariate_estimates
+from get_draws.api import get_draws
+import scipy.stats
+import scipy.integrate as integrate
+import matplotlib.pyplot as plt
+
+def generate_coverage_parameter_draws(df):
+    """This function is used to generate 1000 draws of nutrient/vehicle coverage parameters based on
+    the mean value and confidence intervals. This function assumes a normal distribution of uncertainty
+    within the confidence interval centered around the mean and is truncated at the bounds of 0 and 100%"""
+    data_frame = df.copy()
+    np.random.seed(11)
+    for i in list(range(0,1000)):
+        data_frame[f'draw_{i}'] = scipy.stats.truncnorm.rvs(data_frame.a,
+                                                            data_frame.b,
+                                                            data_frame.value_mean,
+                                                            data_frame.value_std) / 100
+    data_frame = (data_frame
+                  .set_index(['location_id'])
+                  .drop(columns=[c for c in data_frame.columns if 'draw' not in c
+                                and c not in ['location_id','value_description']]))
+    return data_frame
+
+
+def generate_overall_coverage_rates(filepath,

```

Comment:
Oh oops I didn't get back to you on this @alibow ! I took this function and broke it up into a few sub-functions that do each step of this, and allow me to pull out some interim tables before it generates the full location-year-coverage_level table.

I could wait until this is merged, and then PR off of this with those? or open up a new PR right now so you can see, and edit it once this branch is merged

Reply:
Ah, I see! I think we can both just merge our branches without breaking anything and then afterwards I will update them all to your easier to follow updated version if that sounds good?